iree/turbine/kernel/wave/promotion.py

```diff
@@ -34,11 +34,21 @@ def apply_padding(
     )
 
 
-def apply_promotion_pattern(custom_node: Read | Write, allocate_node: Allocate):
+def apply_promotion_pattern(
```

Comment:
Could you add your comment from the PR in here? Would be helpful to understand what is going on.

Reply:
done!